### PR TITLE
JAMES-3734: Document tools and performance testing result for James

### PR DESCRIPTION
**Pull request by quantranhong1999**

### CODE REVIEW

File: server/apps/distributed-app/docs/modules/ROOT/pages/benchmark/james-benchmark.adoc [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+* JMAP draft: `org.apache.james.gatling.simulation.jmap.draft.PlatformValidationSimulation`
+* JMAP rfc8621: `org.apache.james.gatling.simulation.jmap.rfc8621.PushPlatformValidationSimulation`
+
+=== Base performance result
+
+A sample IMAP performance testing result (PlatformValidationSimulation):
+
+image::james-imap-base-performance.png[]
+
+If you get a IMAP performance far below this base performance, you should consider having performance issues.
```
Comment (chibenwa) [inline server/apps/distributed-app/docs/modules/ROOT/pages/benchmark/james-benchmark.adoc:94]:
```suggestion
If you get a IMAP performance far below this base performance, you should consider investigating for performance issues.
```

File: server/apps/distributed-app/docs/modules/ROOT/pages/benchmark/james-benchmark.adoc [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+performance issues.
+
+== Sample deployment topology
+
+We deploy a sample topology of Distributed James with these following components:
+
+- Apache Cassandra 4 as main database
+- OpenDistro 1.13.1 as search engine
+- RabbitMQ 3.8.17 as message queue
+- OVH Swift S3 as an object storage
```
Comment (chibenwa) [inline server/apps/distributed-app/docs/modules/ROOT/pages/benchmark/james-benchmark.adoc:20]: Please specofy servers counts, CPU, memory. Specify also the counts of James PODs, CPU, memory.

File: server/apps/distributed-app/docs/modules/ROOT/pages/benchmark/james-benchmark.adoc [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+* Setup James Gatling with `sbt` build tool
+
+* Configure the `Configuration.scala` to point to your Distributed James IMAP/JMAP server(s). For more configuration details, please read
+https://github.com/linagora/james-gatling#readme[James Gatling Readme].
+
+* Run the performance testing simulation:
+----
+$ sbt
+> gatling:testOnly SIMULATION_FQDN
```
Comment (chibenwa) [inline server/apps/distributed-app/docs/modules/ROOT/pages/benchmark/james-benchmark.adoc:77]: Can we specify this FQDN ?

File: server/apps/distributed-app/docs/modules/ROOT/pages/benchmark/james-benchmark.adoc [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+:navtitle: James benchmarks
+
+This document provides benchmark methodology and basic performance of Distributed James as a basis for a James administrator who
+can test and evaluate if his Distributed James is performing well.
+
+It includes:
+
+* A sample Distributed James deployment topology
+* Propose benchmark methodology and base performance result. This aims to help operators quickly identify
+performance issues.
```
Comment (chibenwa) [inline server/apps/distributed-app/docs/modules/ROOT/pages/benchmark/james-benchmark.adoc:11]:
```suggestion
* A sample Distributed James deployment topology
* Propose benchmark methodology 
 * Sample performance results. 
 
 This aims to help operators quickly identify performance issues.
performance issues.
```

File: server/apps/distributed-app/docs/modules/ROOT/pages/benchmark/db-benchmark.adoc [hunk 5]
```diff
@@ -14,9 +14,9 @@ performance issues and compliance of their databases.
 
 We deploy a sample topology of Distributed James with these following databases:
 
-- Apache Cassandra 4 as main database
-- OpenDistro 1.13.1 as search engine
-- RabbitMQ 3.8.17 as message queue
+- Apache Cassandra 4 as main database: 3 nodes, each node has 8 OVH vCores CPU and 30 GB memory limit.
```
Comment (chibenwa) [inline server/apps/distributed-app/docs/modules/ROOT/pages/benchmark/db-benchmark.adoc:17]: Model (B2-30) also?

File: server/apps/distributed-app/docs/modules/ROOT/pages/benchmark/james-benchmark.adoc [hunk 6]
```diff
@@ -77,7 +80,9 @@ $ sbt
 > gatling:testOnly SIMULATION_FQDN
 ----
 
-We did provide a lot of simulations. You can have a look and choose the suitable simulation, or you can even customize your simulation also!
+In there: `SIMULATION_FQDN` is fully qualified class name of a performance test simulation.
+
+We did provide a lot of simulations in `org.apache.james.gatling.simulation` path. You can have a look and choose the suitable simulation, or you can even customize your simulation also!
```
Comment (chibenwa) [inline server/apps/distributed-app/docs/modules/ROOT/pages/benchmark/james-benchmark.adoc:85]:
```suggestion
We did provide a lot of simulations in `org.apache.james.gatling.simulation` path.You can have a look and choose the suitable simulation.  `sbt gatling:testOnly org.apache.james.gatling.simulation.imap.PlatformValidationSimulation` is a good starting point. Or you can even customize your simulation also!
```

**Review by chibenwa — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Pull request merged**